commit bee67c6b7129a6764c964a93e35e33b7cf2b35ea
Author: spiff <spiff@xbmc.org>
Date:   Wed Jan 18 16:33:47 2012 +0100

    fixed: don't deref null pointers, not even in release builds

diff --git a/xbmc/addons/AddonInstaller.cpp b/xbmc/addons/AddonInstaller.cpp
index dc5c366..a7d96c0 100644
--- a/xbmc/addons/AddonInstaller.cpp
+++ b/xbmc/addons/AddonInstaller.cpp
@@ -262,6 +262,9 @@ void CAddonInstaller::InstallFromXBMCRepo(const set<CStdString> &addonIDs)
 bool CAddonInstaller::CheckDependencies(const AddonPtr &addon)
 {
   assert(addon.get());
+  if (!addon)
+    return false;
+
   ADDONDEPS deps = addon->GetDeps();
   CAddonDatabase database;
   database.Open();